python/lsst/analysis/tools/tasks/makeMetricTable.py

```diff
@@ -112,62 +125,71 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
         """
 
         inputs = butlerQC.get(inputRefs)
-        skymap = inputs["skymap"]
-
-        # Make a list of tracts
-        tracts = []
+        if "skymap" in inputs:
+            skymap = inputs["skymap"]
+        else:
+            skymap = None
+
+        # Extract the info from the dataIds that is needed
+        # to populate the requested columns.
+        fields = self.config.dataIdFieldsToIncludeAsColumns
+        dataIdInfo = []
         for data in inputRefs.data:
-            tracts.append(data.dataId["tract"])
+            dataIdInfo.append({field: data.dataId[field] for field in fields})
 
         metricBundles = []
         for inputHandle in inputs["data"]:
             metricBundles.append(inputHandle.get())
 
-        outputs = self.run(tracts, metricBundles, skymap)
+        outputs = self.run(dataIdInfo, metricBundles, skymap)
         butlerQC.put(outputs, outputRefs)
 
-    def run(self, tracts, metricBundles, skymap):
-        """Take the metric bundles and expand them out,
-        add the tract corner information and then make
-        a table of the information.
+    def run(self, dataIdInfo, metricBundles, skymap):
+        """Take the metric bundles and expand them out, then make a table of
+        the information. Add tract corner information if the bundles are
+        tract-level.
 
         Parameters
         ----------
-        tracts : `list`
-            A list of the tracts that the metricBundles cover
+        dataIdInfo : `list`
+            A list of dicts that hold information extracted from the metric
+            bundle dataIds.
         metricBundles : `list` of
             `lsst.analysis.tools.interfaces._metricMeasurementBundle.MetricMeasurementBundle`
         skymap : `lsst.skymap`
 
         Returns
         -------
-        tMetrics : `pipe.base.Struct` containing `astropy.table.Table`
+        metricTableStruct : `pipe.base.Struct` containing `astropy.table.Table`
         """
 
         # Make an initial dict of the columns needed
         metricsDict = {}
-        tract = tracts[0]
-        metricsDict["tract"] = [tract]
-        corners = self.getTractCorners(skymap, tract)
-        metricsDict["corners"] = [corners]
+        for key, value in dataIdInfo[0].items():
```

Comment:
Done. I've also added a check to make sure that metricBundles also has at least one item (for the same reason). Both reaise a NoWorkFound error if either lists are empty. It shouldn't get to this point if run is called via the task's `runQuantum` method, but I agree that folk may want to use the `run` method directly, making this check and subsequent message useful.

Reply:
I've added a `try:/except: IndexError` to check that each of `dataIdInfo` and `metricBundles` can each be indexed. It shouldn't ever get to that if the `run` method is called by `runQuantum`, but I agree that we should have a check in case someone does a standalone `run` call from elsewhere.